src/main/java/com/ericsson/ei/rules/MatchIdRulesHandler.java

```diff
@@ -67,7 +73,10 @@ public String replaceIdInRules(String matchIdString, String id) {
         if (matchIdString.contains(replacementMarker)) {
             return matchIdString.replace(replacementMarker, id);
         } else {
-            return null;
+            String errorMessage = String.format("MatchIdRules: %s does not contain the rules.replacement.marker: %s", matchIdString, replacementMarker);
+            ReplacementMarkerException exception = new ReplacementMarkerException(errorMessage);
+            LOGGER.warn("", exception);
+            throw exception;
```

Comment:
Where is the test for this?

Reply:
Added a test, code has changed significantly though.